Tools/ardupilotwaf/configuration_helper.py

```diff
@@ -0,0 +1,78 @@
+"""
+WAF Tool that helps on configuration parameter, creating the AP_Config
+header file.
+
+This tool needs compiler_c and compiler_cxx to be loaded, make sure
+you load them before this tool.
+
+Example::
+    def configure(cfg):
+        cfg.load('configuration_helper')
+"""
+
+def configure(cfg):
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isfinite(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISFINITE",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isinf(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISINF",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isnan(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISNAN",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <math.h>
+                  #include <cmath>
+
+                  using std::isfinite;
+
+                  int main() {
+                    return isfinite((double)1);
+                  }''',
+                  define_name="NEED_CMATH_ISFINITE_STD_NAMESPACE",
+                  cxxflags="-std=gnu++11",
```

Comment:
Here is the diff, by the way:
```diff
diff --git a/Tools/ardupilotwaf/boards.py b/Tools/ardupilotwaf/boards.py
index fe01d35..b67b702 100644
--- a/Tools/ardupilotwaf/boards.py
+++ b/Tools/ardupilotwaf/boards.py
@@ -52,6 +52,8 @@ class Board:
             else:
                 cfg.env[k] = val
 
+        cfg.load('configuration_helper')
+
     def configure_env(self, cfg, env):
         # Use a dictionary instead of the convetional list for definitions to
         # make easy to override them. Convert back to list before consumption.
diff --git a/Tools/ardupilotwaf/toolchain.py b/Tools/ardupilotwaf/toolchain.py
index 8901873..e09728d 100644
--- a/Tools/ardupilotwaf/toolchain.py
+++ b/Tools/ardupilotwaf/toolchain.py
@@ -119,7 +119,6 @@ def _filter_supported_cxx_compilers(*compilers):
 def configure(cfg):
     if cfg.env.TOOLCHAIN == 'native':
         cfg.load('compiler_cxx compiler_c')
-        cfg.load('configuration_helper')
         return
 
     _filter_supported_c_compilers('gcc', 'clang')
@@ -129,8 +128,6 @@ def configure(cfg):
     cfg.msg('Using toolchain', cfg.env.TOOLCHAIN)
     cfg.load('compiler_cxx compiler_c')
 
-    cfg.load('configuration_helper')
-
     if cfg.env.COMPILER_CC == 'clang':
         cfg.env.CFLAGS += cfg.env.CLANG_FLAGS
         cfg.env.LINKFLAGS_cprogram += cfg.env.CLANG_FLAGS
```